tutorials/AgileRL/agilerl_maddpg.py

```diff
@@ -0,0 +1,239 @@
+        "h_size": [32, 32],  # Network hidden size
+        "c_size": [3, 32],  # CNN channel size
+        "k_size": [(1, 3, 3), (1, 3, 3)],  # CNN kernel size
+        "s_size": [2, 2],  # CNN stride size
+        "normalize": True,  # Normalize image from range [0,255] to [0,1]
+    }
+
+    # Define the initial hyperparameters
+    INIT_HP = {
+        "POPULATION_SIZE": 2,
+        "ALGO": "MADDPG",  # Algorithm
+        # Swap image channels dimension from last to first [H, W, C] -> [C, H, W]
+        "CHANNELS_LAST": True,
+        "BATCH_SIZE": 8,  # Batch size
+        "LR": 0.01,  # Learning rate
+        "GAMMA": 0.95,  # Discount factor
+        "MEMORY_SIZE": 10000,  # Max memory buffer size
+        "LEARN_STEP": 5,  # Learning frequency
+        "TAU": 0.01,  # For soft update of target parameters
+    }
+
+    # Define the space invaders environment as a parallel environment
+    env = space_invaders_v2.parallel_env()
+    if INIT_HP["CHANNELS_LAST"]:
+        # Environment processing for image based observations
+        env = ss.frame_skip_v0(env, 4)
+        env = ss.clip_reward_v0(env, lower_bound=-1, upper_bound=1)
+        env = ss.color_reduction_v0(env, mode="B")
+        env = ss.resize_v1(env, x_size=84, y_size=84)
+        env = ss.frame_stack_v1(env, 4)
+    env.reset()
+
+    # Configure the multi-agent algo input arguments
+    try:
+        state_dim = [env.observation_space(agent).n for agent in env.agents]
+        one_hot = True
+    except Exception:
+        state_dim = [env.observation_space(agent).shape for agent in env.agents]
+        one_hot = False
+    try:
+        action_dim = [env.action_space(agent).n for agent in env.agents]
+        INIT_HP["DISCRETE_ACTIONS"] = True
+        INIT_HP["MAX_ACTION"] = None
+        INIT_HP["MIN_ACTION"] = None
+    except Exception:
+        action_dim = [env.action_space(agent).shape[0] for agent in env.agents]
+        INIT_HP["DISCRETE_ACTIONS"] = False
+        INIT_HP["MAX_ACTION"] = [env.action_space(agent).high for agent in env.agents]
+        INIT_HP["MIN_ACTION"] = [env.action_space(agent).low for agent in env.agents]
+
+    # Pre-process image dimensions for pytorch convolutional layers
+    if INIT_HP["CHANNELS_LAST"]:
+        state_dim = [
+            (state_dim[2], state_dim[0], state_dim[1]) for state_dim in state_dim
+        ]
+
+    # Append number of agents and agent IDs to the initial hyperparameter dictionary
+    INIT_HP["N_AGENTS"] = env.num_agents
+    INIT_HP["AGENT_IDS"] = env.agents
+
+    # Create a population ready for evolutionary hyper-parameter optimisation
+    pop = initialPopulation(
+        INIT_HP["ALGO"],
+        state_dim,
+        action_dim,
+        one_hot,
+        NET_CONFIG,
+        INIT_HP,
+        population_size=INIT_HP["POPULATION_SIZE"],
+        device=device,
+    )
+
+    # Configure the multi-agent replay buffer
+    field_names = ["state", "action", "reward", "next_state", "done"]
```

Comment:
Hi, small nitpick, I assume this is the name for the element in the replay buffer? Does changing this to `termination` meaningfully modify anything with the learning library? If not, could we proceed with the change?